pip/commands/list.py

```diff
@@ -169,6 +181,15 @@ def get_uptodate(self, packages, options):
             if dist.latest_version == dist.parsed_version
         ]
 
+    def get_nodeps(self, packages, options):
+        installed_packages = [
+            dist for dist in self.iter_packages_latest_infos(packages, options)]
+        all_dependencies = []
+        for dist in installed_packages:
+            all_dependencies.extend(dist.requires())
+        dep_keys = set(dep.key for dep in all_dependencies)
+        return set(pkg for pkg in installed_packages if pkg.key not in dep_keys)
```

Comment:
IIUC `installed_packages` and `deps_keys` contain objects of different types. `dist.requires()` returns `Requirement` objects and `dist` has other type.